src/components/accessories/admin/users/editUser/EditUser.tsx

```diff
@@ -0,0 +1,64 @@
+import { CircularProgress } from "@mui/material";
+import { useAppDispatch, useAppSelector } from "libraries/hooks/redux";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { Navigate } from "react-router-dom";
+
+import { PATHS } from "../../../../../consts";
+import { UserDTO } from "../../../../../generated";
+import { getUserGroups } from "../../../../../state/usergroups";
+import { getUsers, updateUser, updateUserReset } from "../../../../../state/users";
+import { EditUserForm } from "./EditUserForm";
+
+export const EditUser = () => {
+  const dispatch = useAppDispatch();
+  const { id } = useParams();
+  const [user, setUser] = useState<UserDTO | null>();
+  const [userNotFound, setUserNotFound] = useState<boolean>(false);
+  const { isLoading, hasSucceeded, hasFailed, error } = useAppSelector(
+    (state) => state.users.update
+  );
+  const users = useAppSelector((state) => state.users.userList);
```

Comment:
I think you should pass the user to edit in the router state or send a request to fetch the user to edit using his username (fetch only this user, not all users). 
Here you're fetching all users and this evolves two problems:
- If the endpoint to get all users is paginated, you might not find the user even though he exists.
- It is not the most efficient way to go about it.

Reply:
Passing the user in the state is not a good practice because it prevent from using the urls to navigate.
I am not confortable with fetching a single user, because it would put a wrong value in the state:

`state.users.userList` would then equal `{username: "john", password: "hello"}` or maybe `[{username: "john", password: "hello"}]`  which are both wrong because `state.users.userList` should be `null` or contain the actual user list. Abiding to this best practice opens the ability to do optimistic ui.

single user data should be stored in a format looking like: `state.users.user.john` where the state can be accessed directly by id.

More readings on this:
https://redux.js.org/usage/structuring-reducers/basic-reducer-structure
and https://redux.js.org/usage/structuring-reducers/normalizing-state-shape

wdyt?

cc @SteveGT96

Reply:
Sorry @gasp , I was busy with something else and I completely forgot to check your PR from time to time.

You can still create another state, maybe `state.users.getByid` or `state.users.getByUsername`. That way, you could dispatch an action that fetch a single user using his id or username, store the result it the state you'll have created, so that, later you can query it. 
Consider the first problem I mentioned 
> If the endpoint to get all users is paginated, you might not find the user even though he exists
Because if there are more that a hundred user, I don't think we're going to fetch them all at once. If the endpoint we use to fetch users is not yet paginated, we should paginate it.

Reply:
100 hundredu user is less than 100 kb
If we add pagination, we will have to rework entirely table filtering.

Reply:
I still believe that here, you have to get a single user, not all the user in the app.

Reply:
Hi @gasp ! Thanks for the efforts.
I don't think takink the user to update from the user list could be the best way(that could be error prone and not really easy to maintain).
That said, when the user click on the edit button then you navigate to /admin/users/:username/edit(or something like) and in the EditUser component, you load the the user matching the username param in the state(state.users.[user | getByUsername]).

Why that pattern ?

Modern application acts that way and considers it as a best practice.
- React router for example gives the possibility to fetch the user before rendering the target component(using data loader pattern, put in our case, we're already using redux).
- Angular for example has a built in approch using resolver guards.